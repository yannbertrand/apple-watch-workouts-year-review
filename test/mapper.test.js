const { mapWorkoutAttributes } = require('../lib/mapper')

const sampleWorkout = require('./samples/apple-workout.json')

describe('mapWorkoutAttributes(attributes)', () => {
  describe('with non existing workoutActivityType', () => {
    it('should throw an Error', () => {
      const sampleAttributes = {
        ...sampleWorkout,
        workoutActivityType: 'HKWorkoutActivityTypeSleeping',
      }

      expect(() => mapWorkoutAttributes(sampleAttributes)).toThrow()
    })
  })

  describe('with existing workoutActivityType', () => {
    let result

    beforeEach(() => {
      result = mapWorkoutAttributes(sampleWorkout)
    })

    it('should set an "id" key', () => {
      expect(result.id).toMatchSnapshot()
    })

    it('should set a "type" key', () => {
      expect(result.type).toMatchSnapshot()
    })

    it('should set a "duration" float key', () => {
      expect(result.duration).toMatchSnapshot()
    })

    it('should set a "totalDistance" float key', () => {
      expect(result.totalDistance).toMatchSnapshot()
    })

    it('should set a "totalEnergyBurned" float key', () => {
      expect(result.totalEnergyBurned).toMatchSnapshot()
    })

    it('should set a "sourceName" key', () => {
      expect(result.sourceName).toMatchSnapshot()
    })

    it('should set a "sourceVersion" key', () => {
      expect(result.sourceVersion).toMatchSnapshot()
    })

    it('should set a "creationDate" ISO date key', () => {
      expect(result.creationDate).toMatchSnapshot()
    })

    it('should set a "startDate" ISO date key', () => {
      expect(result.startDate).toMatchSnapshot()
    })

    it('should set a "endDate" ISO date key', () => {
      expect(result.endDate).toMatchSnapshot()
    })
  })
})
