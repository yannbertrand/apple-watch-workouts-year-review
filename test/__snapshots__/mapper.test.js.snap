// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "creationDate" ISO date key 1`] = `"2019-12-17T19:57:46.000Z"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "duration" float key 1`] = `6.115610067049662`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "endDate" ISO date key 1`] = `"2019-12-17T19:57:46.000Z"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "sourceName" key 1`] = `"Yann's AppleÂ Watch"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "sourceVersion" key 1`] = `"5.2.1"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "startDate" ISO date key 1`] = `"2019-12-17T19:51:39.000Z"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "totalDistance" float key 1`] = `1.759895436402728`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "totalEnergyBurned" float key 1`] = `32.742351690196`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set a "type" key 1`] = `"cycling"`;

exports[`mapWorkoutAttributes(attributes) with existing workoutActivityType should set an "id" key 1`] = `"2019-12-17 20:51:39 +01002019-12-17 20:57:46 +0100HKWorkoutActivityTypeCycling6.1156100670496621.75989543640272832.742351690196"`;
